### PR TITLE
feat: Polygon finalizer

### PR DESCRIPTION
**Pull request by nicholaspai**
This is still just a script that can be run with `ts-node ./src/finalizer/index.ts --wallet mnemonic` which is janky and manual for now but its confirmed to work with polygon and at least prints out transaction hashes for arbitrum

### CODE REVIEW

File: src/finalizer/utils/arbitrum.ts [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+  winston,
+  convertFromWei,
+  groupObjectCountsByTwoProps,
+} from "../../utils";
+import { L2ToL1MessageWriter, L2ToL1MessageStatus, L2TransactionReceipt, getL2Network } from "@arbitrum/sdk";
+import { MessageBatchProofInfo } from "@arbitrum/sdk/dist/lib/message/L2ToL1Message";
+import { TokensBridged } from "../../interfaces";
+import { HubPoolClient } from "../../clients";
+
+export async function finalizeArbitrum(
```
Comment (mrice32) [inline src/finalizer/utils/arbitrum.ts:14]: Unclear how much of this was copied from v1 (hopefully most of it!). Might be worth commenting in the PR what sections of code are new.

**Review by mrice32 — APPROVED** [review]
Would be really nice to see multicall and simulation mode added here, but those may be tricky given how these SDKs work.

This looks good, and I think it's relatively low risk code.

The biggest concern I have is hardening it to make it less noisy than the Across V1 finalizers.

LGTM!

File: src/finalizer/utils/arbitrum.ts (hunk 1 above)
Comment (nicholaspai) [reply src/finalizer/utils/arbitrum.ts:14]: Its all new actually, we don't use updated SDK's in new code

**Comment by nicholaspai** [comment]
@mrice32 we can't use multicall easily since the contracts we're calling are all different. We could go through a DS proxy...

**Pull request merged**